pandas/_libs/algos.pyx

```diff
@@ -237,34 +245,84 @@ def groupsort_indexer(const int64_t[:] index, Py_ssize_t ngroups):
     return indexer, counts
 
 
+cdef inline Py_ssize_t swap(numeric *a, numeric *b) nogil:
+    cdef:
+        numeric t
+
+    # cython doesn't allow pointer dereference so use array syntax
+    t = a[0]
+    a[0] = b[0]
+    b[0] = t
+    return 0
+
+
+cdef inline numeric kth_smallest_c(numeric* arr, Py_ssize_t k, Py_ssize_t n) nogil:
+    """
+    Compute the kth smallest value in an array
+
+    Parameters
+    ----------
+    arr: numeric* arr
```

Comment:
trailing "arr" not needed?

space before the colon in Parameters section

Reply:
Done